module/modmanager/manager_test.go

```diff
@@ -1322,3 +1322,34 @@ func TestBadModuleFailsFast(t *testing.T) {
 
 	test.That(t, err.Error(), test.ShouldContainSubstring, "module test-module exited too quickly after attempted startup")
 }
+
+func TestModuleDiscoverRegistered(t *testing.T) {
+	ctx := context.Background()
+	logger := logging.NewTestLogger(t)
+
+	modPath := rtestutils.BuildTempModule(t, "module/testmodule")
+
+	modCfg := config.Module{
+		Name:    "test-module",
+		ExePath: modPath,
+	}
+
+	parentAddr := setupSocketWithRobot(t)
+
+	mgr := setupModManager(t, ctx, parentAddr, logger, modmanageroptions.Options{UntrustedEnv: false})
+
+	err := mgr.Add(ctx, modCfg)
+	test.That(t, err, test.ShouldBeNil)
+
+	// The helper model implements actual (foobar) discovery
+	reg, ok := resource.LookupRegistration(generic.API, resource.NewModel("rdk", "test", "helper"))
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, reg, test.ShouldNotBeNil)
+
+	// Check that the Discover function is registered.
+	test.That(t, reg.Discover, test.ShouldNotBeNil)
+
+	res, err := reg.Discover(ctx, logger)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, res, test.ShouldNotBeNil)
```

Comment:
let's test that the response resembles [the mock discover response you added](https://github.com/viamrobotics/rdk/pull/4410/files#diff-e7ba4c8e22457875e841fc23dd1efa585bed7f88fcc37e260886e80a7edb9ab7R52-R54).

Reply:
Done